test/StrategyManager/invest.test.ts

```diff
@@ -180,26 +227,89 @@ describe('StrategyManager#invest', () => {
         ))
 
         return {
-            strategyId: _strategyId,
-            inputToken: stablecoin,
-            inputAmount: _amount,
-            inputTokenSwap: '0x',
+            strategyId,
+            inputTokenSwap,
             dcaSwaps: dcaInvestments.map(_ => '0x'),
             vaultSwaps: vaultInvestments.map(_ => '0x'),
             liquidityZaps,
             buySwaps: [], // TODO
             investorPermit,
-            strategistPermit: _strategistSubscribed ? permitAccount0 : expiredPermitAccount0,
+            strategistPermit: strategistSubscribed ? permitAccount0 : expiredPermitAccount0,
         }
     }
 
+    async function getInvestParams(investor: Signer, {
+        amount = amountToInvest,
+        inputToken = stablecoinPriced,
+        _strategyId = strategyId,
+        investorSubscribed = true,
+        strategistSubscribed = true,
+    }: {
+        amount?: bigint,
+        inputToken?: ERC20Priced,
+        _strategyId?: bigint,
+        investorSubscribed?: boolean,
+        strategistSubscribed?: boolean,
+    } = {
+        amount: amountToInvest,
+        inputToken: stablecoinPriced,
+        _strategyId: strategyId,
+        investorSubscribed: true,
+        strategistSubscribed: true,
+    }): Promise<StrategyInvestor.InvestParamsStruct> {
+        return {
+            ...await getBaseInvestParams(
+                _strategyId,
+                amount,
+                inputToken,
+                investor,
+                { investorSubscribed, strategistSubscribed },
+            ),
+            inputToken: inputToken.address,
+            inputAmount: amount,
+        }
+    }
+
+    async function getInvestNativeParams(
+        investor: Signer,
+        {
+            amount = amountToInvest,
+            _strategyId = strategyId,
+            investorSubscribed = true,
+            strategistSubscribed = true,
+        }: Parameters<typeof getInvestParams>[1] = {
+            amount: amountToInvest,
+            _strategyId: strategyId,
+            investorSubscribed: true,
+            strategistSubscribed: true,
+        },
+    ): Promise<StrategyInvestor.InvestNativeParamsStruct> {
+        return getBaseInvestParams(
+            _strategyId,
+            amount,
+            wethPriced,
+            investor,
+            { investorSubscribed, strategistSubscribed, isNativeInvest: true },
+        )
+    }
```

Comment:
functions with this many optional arguments are extremely hard to read, please simplify it and only use optional arguments if it really makes sense, I always feel like its better to be explicit when calling functions anyway, especially on tests